File: with_elixir/test/day1_test.exs
defmodule Day1Test do
  use ExUnit.Case
  doctest Day1

  import ReadInput

  test "part1..." do
    assert Day1.part1(read_input("day1part1sample")) == 11
  end

  test "part2..." do
    assert Day1.part2(read_input("day1part1sample")) == 31
  end
end

File: 2024/with_node/src/day1.test.ts
import assert from "node:assert";
import test from "node:test";

import { readInput } from "./readInput.ts";
import { part1 as day1part1, part2 as day1part2 } from "./day1.ts";

test("day1...", async (t) => {
  await t.test("part1 says...", () => {
    assert.strictEqual(day1part1(readInput("day1sample")), 11);
  });

  await t.test("part2 says...", () => {
    assert.strictEqual(day1part2(readInput("day1sample")), 31);
  });
});
